app/models/user.rb

```diff
@@ -9,4 +9,47 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :friendships
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
+
+  # Returns an array with the friends
+  def friends
```

Comment:
- **[OPTIONAL]** [ ] `friends` can be fetched with `has_many through` association:
```ruby
has_many :confirmed_friends, -> { where(friendships: { confirmed: true }) }, through: :friendships, source: :friend
```
This is better because active record querying is faster than using ruby enumerables.

Reply:
Solved, thank you very much @mosaaleb